libpod/runtime.go

```diff
@@ -199,7 +199,7 @@ var (
 			"PATH=/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin",
 		},
 		CgroupManager:         SystemdCgroupsManager,
-		HooksDir:              hooks.DefaultDir,
+		HooksDir:              []string{hooks.DefaultDir, hooks.OverrideDir},
```

Comment:
This is changing our semantics a bit, since libpod has never used `OverrideDir`.  It landed in #686 for CRI-O, where it is [only used when the user is not overriding their hook dir for testing][1].  I'm not particularly comfortable with either approach as the default; I'd rather have an empty `HooksDir` (or `HooksDirs`?) array as the default in both libpod and CRI-O, with directories added explicitly via configuration files or command-line options.  In that case, we could drop `dirNotExistFatal` and *always* be fatal on dir-not-exist.  That approach will break backwards compat, so we'd want to phase it in slowly by whining whenever we loaded hooks from a default-but-not-explicitly-configured directory for a few versions before dropping the default directories.  Adding more hard-coded hook directories without a way to remove them seems to be moving in the wrong direction.

[1]: https://github.com/kubernetes-sigs/cri-o/blob/v1.11.4/lib/container_server.go#L177-L180

Reply:
We only use this for override.  The reason this was caught is people are trying to get hooks to work with CoreOS. You are not allowed to write to /usr/share... on CoreOS, so they wanted podman to also look at /etc, where users and third parties can write.

Reply:
> The reason this was caught is people are trying to get hooks to work with CoreOS.

But with your approach here, say you're just providing a new, hard-coded way to add hooks.  Say you have a hook in `/usr/share/...` that you want to drop.  With an "explicitly configured hook dirs" approach, you just drop that directory from your configuration and point libpod at a directory with the hooks you want.  This favors users who want to control their hooks without worrying about drifting system defaults.  It also makes it easier to migrate libpod defaults, because users have explicit local configuration that they can migrate independently, vs. having defaults baked into libpod (at which point they're basically frozen in).

With your "hard-code an override directory", you'd have to add a covering entry with the same filename to your override directory but with no-op content (maybe we should make `Hook.Hook` an optional pointer to support this?).  This is good for users that want to track the system defaults with minor alterations and don't mind the occasional breakage.

Reply:
Well I would want to wait for a real user to request this before implementing it. Theoretically you are correct.  But for now, we have a real use case for /etc/containers/hooks.

Reply:
> Well I would want to wait for a real user to request this before implementing it.

Now that I work on openshift/installer, do I get to be a "real user"? ;)

> But for now, we have a real use case for /etc/containers/hooks.

Making `--hooks-dir-path` public and documenting/supporting `hooks_dir` in the libpod configuration file would allow for this use case too.  And I prefer that approach, since it slants towards stability in the face of system and libpod changes.